include/boost/heap/detail/stable_heap.hpp

```diff
@@ -173,19 +173,21 @@ struct heap_base:
     heap_base(heap_base && rhs) BOOST_NOEXCEPT_IF(boost::is_nothrow_move_constructible<Cmp>::value):
 #ifndef BOOST_MSVC
         Cmp(std::move(static_cast<Cmp&>(rhs))),
-#else
-        cmp_(std::move(rhs.cmp_)),
 #endif
         size_holder_type(std::move(static_cast<size_holder_type&>(rhs)))
```

Comment:
is it possibly to move the initialisation of `size_holder_type` up? or would this result in a similar warning?

Reply:
I assume that this would result in a similar warning for gcc and clang.

Reply:
One could however duplicate the `size_holder_type` in both cases of the `ifdef`, i.e.
```c++
#ifndef BOOST_MSVC
        Cmp(std::move(static_cast<Cmp&>(rhs))),
        size_holder_type(std::move(static_cast<size_holder_type&>(rhs)))
#else
        cmp_(std::move(rhs.cmp_)),
        size_holder_type(std::move(static_cast<size_holder_type&>(rhs)))
#endif
```

Reply:
Another alternative might be to let `struct heap_base` derive first from `size_holder<...>` and then from `Cmp` for non-MSVC compilers. Not sure, though, whether this will have implications somewhere else.

Reply:
Then we could leave the constructors unchanged, i.e.
```c++
#ifndef BOOST_MSVC
        Cmp(std::move(static_cast<Cmp&>(rhs))),
#else
        cmp_(std::move(rhs.cmp_)),
#endif
        size_holder_type(std::move(static_cast<size_holder_type&>(rhs)))
```

Reply:
probably not worth the effort to silence a small warning ...

thanks a lot for the PR!

Reply:
You are welcome!